magento/magento2
# Refactor moveFileFromTmp to not use a static method call

### Description (*)

Currently the class `\Magento\Catalog\Model\ImageUploader` has a static call to `Uploader::getNewFileName`

In this PR I plan to:
 - Create new class for the method `getNewFileName`,
 - Refactor the `\Magento\Catalog\Model\ImageUploader` class to use this new class,

### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Upload an image to a category with a unique name and check that name has not been changed,
2. Upload an image to a category with a nonunique name and check that the name has been updated to be unique,

### Questions or comments


I am not too sure on the naming of the new class but would be happy to update it if needed.
I am also not sure about if I need an optional injection of my new class to the `ImageUploader` class.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#29598: Refactor moveFileFromTmp to not use a static method call
